analyses/cell-type-ewings/exploratory_analysis/04-infercnv.Rmd

```diff
@@ -0,0 +1,485 @@
+
+# output from running infercnv 
+results_dir <- file.path(module_base, "results", "infercnv", params$library_id)
+
+all_results_dir_list <- list(
+  no_ref = file.path(results_dir, "no_reference"),
+  with_ref = file.path(results_dir, "endothelial_reference")
+)
+
+# png ref files 
+png_files <- all_results_dir_list |>
+  purrr::map_chr(\(dir) file.path(dir, glue::glue("{params$library_id}_infercnv.png")))
+
+png_files |> 
+  # make sure png files exists
+  purrr::map(\(file)
+    stopifnot("Missing PNG file" = file.exists(file))
+  )
+
+# with ref cnv metadata file 
+cnv_metadata_file <- file.path(results_dir, "endothelial_reference", glue::glue("{params$library_id}_cnv-metadata.tsv"))
+
+# full infercnv object with information on genes/ chr to use later 
+final_cnv_obj_file <- file.path(results_dir, "endothelial_reference", glue::glue("{params$library_id}_cnv-obj.rds"))
+```
+
+```{r}
+# read in sce object
+sce <- readr::read_rds(sce_file)
+
+# read in cnv metadata 
+# use read.delim instead of readr because we want the rownames to be read in properly 
+cnv_metadata_df <- read.delim(cnv_metadata_file) |> 
+  tibble::rownames_to_column("barcodes")
+
+# read in tumor normal classifications 
+manual_classifications_df <- readr::read_tsv(classifications_file)
+
+# read in cnv object 
+final_cnv_obj <- readr::read_rds(final_cnv_obj_file)
+```
+
+
+## InferCNV Heatmap 
+
+### Endothelial cells as reference 
+
+Below is the heatmap produced by `InferCNV` showing the CNV's for each cell using endothelial cells as a reference. 
+Each row is a cell and each column is a gene. 
+The top group contains all cells that were used as the reference group and the bottom is all other cells. 
+The color indicates the relative expression intensity of each gene with the darker colors either indicating a gain of expression or loss of expression compared to the reference cells.
+
+From the [`InferCNV` documentation](https://github.com/broadinstitute/inferCNV/wiki/Interpreting-the-figure): 
+
+> The normal cells in the top heatmap define baseline expression for genes in normal cells. This baseline distribution of normal gene expression is subtracted from both the normal cells from which was defined as well as the tumor cells. Afterwards, the normal cell expression heatmap should be largely devoid of signal, with the exception of certain outlier gene expression values in certain cells. Removing this baseline of normal expression signal from the tumor cells should reveal those chromosomal regions that have significantly more or less expression than the normal cells, highlighting likely amplified or deleted whole chromosomes or large chromosomal regions.
+
+The dendogram on the left side is hierarchical clustering of all non-reference cells. 
+The color annotation next to it indicates subdivisions of the dendogram by cutting it in `k_obs_groups` groups, with the default `k_obs_groups` = 1. 
```

Comment:
Which color annotation is this? There are two, it seems to me, and I can't tell what the rainbow is: it certainly isn't `k_obs_groups = 1`!

Reply:
I'm having trouble understanding the docs here, because it sounds like the bar immediately next to the dendogram (the rainbow one) is the subdivisions of the dendogram. I think there is an error in the docs, because you are correct that is not `k_obs_groups=1`. And the right most bar is supposed to be the original annotation provided.